File: easy-node/frontend/src/types/node.ts
export interface Node {
  id: number;
  name: string;
  status: string;
  base_port: number;
  enabled: boolean;
  peer_key: string;
  config_yml?: string;
  keys_yml?: string;
  health_status: 'unknown' | 'healthy' | 'unhealthy' | 'checking';
  last_health_check?: string;
  health_message?: string;
  response_time: number;
  created_at: string;
  updated_at: string;
}

export interface NodeStatus {
  id: number;
  name: string;
  container_name: string;
  container_status: string;
  base_port: number;
  enabled: boolean;
}

export interface NodeHealthInfo {
  node_id: number;
  node_name: string;
  container_status: string;
  health_status: 'unknown' | 'healthy' | 'unhealthy' | 'checking';
  last_check: string;
  response_time: number;
  health_message: string;
  metrics?: Record<string, any>;
}

export interface HealthStats {
  total_nodes: number;
  healthy_nodes: number;
  unhealthy_nodes: number;
  unknown_nodes: number;
  health_percentage: number;
  avg_response_time: number;
  last_check: string;
}

export interface PerformanceMetrics {
  overview: HealthStats;
  performance: {
    total_nodes: number;
    online_nodes: number;
    avg_response_time: number;
    max_response_time: number;
    min_response_time: number;
    uptime_percentage: number;
  };
  scheduler: {
    running: boolean;
    interval: string;
    next_check: string;
  };
  timestamp: string;
}

export interface CreateNodeRequest {
  name: string;
  base_port: number;
}

export interface UploadNodeRequest extends FormData {
  name: string;
  base_port: number;
  config: File;
  keys: File;
}

File: easy-node/backend/internal/model/node.go
package model

import (
	"gorm.io/gorm"
	"time"
)

type Node struct {
	ID             uint           `json:"id" gorm:"primaryKey"`
	Name           string         `json:"name" gorm:"not null;unique"`
	Status         string         `json:"status" gorm:"default:stopped"`
	BasePort       int            `json:"base_port" gorm:"not null"`
	Enabled        bool           `json:"enabled" gorm:"default:true"`
	PeerKey        string         `json:"peer_key" gorm:"unique;index"`
	ConfigYml      string         `json:"config_yml" gorm:"type:text"`
	KeysYml        string         `json:"keys_yml" gorm:"type:text"`
	
	// 健康检查相关字段
	HealthStatus   string         `json:"health_status" gorm:"default:unknown"`    // unknown, healthy, unhealthy, checking
	LastHealthCheck *time.Time    `json:"last_health_check"`
	HealthMessage  string         `json:"health_message"`
	ResponseTime   float64        `json:"response_time"`                           // 响应时间（毫秒）
	
	CreatedAt      time.Time      `json:"created_at"`
	UpdatedAt      time.Time      `json:"updated_at"`
	DeletedAt      gorm.DeletedAt `json:"deleted_at" gorm:"index"`
}

type NodePortMapping struct {
	NodeID       uint   `json:"node_id"`
	ContainerPort int   `json:"container_port"`
	HostPort     int    `json:"host_port"`
}

func (Node) TableName() string {
	return "nodes"
}